### ISSUE TITLE
Make method for thresholding flexible

### ISSUE THREAD

**Issue by ns-rse** [AFM-SPM/TopoStats#118]
**Is your feature request related to a problem? Please describe.**

Currently the method used for thresholding is fixed using the [Otsu method]() as implemented in [skimage.filters.threshold_otsu](https://scikit-image.org/docs/stable/api/skimage.filters.html?highlight=otsu#skimage.filters.threshold_otsu).

This may not be the optimal/desired method and there are lots of other methods available (see those available at [skimage.filters](https://scikit-image.org/docs/stable/api/skimage.filters.html) for exampes.

**Describe the solution you'd like**

Making the method configurable with a default method that performs well in most instances is desirable.

This can be achieved using a [Factory Design Pattern](https://realpython.com/factory-method-python/) allowing additional filters to be added.

**Comment by bobturneruk** [comment]
Hi @ns-rse - need to bear in mind @alicepyne 's input that having one approach that works for a wide range of input data so as to minimise the need for using tweaking is optimal. Nonetheless this sounds great and I'll have a look.

**Comment by ns-rse** [comment]
Thanks @bobturneruk , I set the default to be that which is currently used `method='otsu'`. I figured adding this makes it easy to test alternative methods further down the line if that is desirable (documentation could even reflect that whilst other methods are available they may not be recommended).

skimage includes local thresholding which may be something to consider. Currently they can't be added with the workflow though as that would require adjusting how thresholds are applied to masking.

But that is getting way ahead, it was simple to add this and it starts making thresholding configurable rather than hard coded.
